### PR TITLE
Fix/refactor query

### PR DESCRIPTION
**Pull request by neewy**
This PR brings small correction of query API description and generalisation of validation section. Also includes #32

### CODE REVIEW

File: source/includes/_queries.md [hunk 1]
```diff
@@ -40,13 +41,22 @@ Each query consists of following things:
     <li> **Query object** might be any of types, described below </li>
 </ul>
 
-**Signatures** contain one or many signatures (ed25519 pubkey + signature):
+**Signatures** contain one or many signatures (ed25519 pubkey + signature).
```
Comment (l4l) [inline source/includes/_queries.md]: I think we can remove note in the parentheses as long as hyperledger/iroha#795 almost merged

File: source/includes/_queries.md [hunk 2]
```diff
@@ -40,13 +41,22 @@ Each query consists of following things:
...
+
+## Validation
+
+The validation for all commands includes:
+
+ * timestamp — it should be in more than time of the peer minus 24 hours
```
Comment (l4l) [inline source/includes/_queries.md]: Maybe "shouldn't be older than 24 hours of the peer's time"?

File: source/includes/_queries.md [hunk 3]
```diff
@@ -40,13 +41,22 @@ Each query consists of following things:
...
+ * signature of query creator — is used for checking the identity of query creator
+ * query counter — is checked to be incremented with every subsequent query from this query creator
+ * roles — depending on the role of query creator, it can be applied to the same account, account in the domain, in the whole system or not allowed at all
```
Comment (l4l) [inline source/includes/_queries.md]: depending on the <s>role of</s> query creator<b>'s role</b>

File: source/includes/_queries.md [hunk 4]
```diff
@@ -56,11 +66,10 @@ Given this Query, in successful case Iroha returns `AccountResponse`, which cont
     <li>quorum — the quantity of signatures to pass within transaction to execute commands for this account </li>
 </ul>
 
-<aside class="warning"> `AccountResponse` should not contain array of permissions, but an array of `Role` object instead. It will be reworked. </aside>
+and also returns the array of account roles.
```
Comment (l4l) [inline source/includes/_queries.md]: Is it fine to separate part of the sentence in such a way?

File: source/includes/_queries.md [hunk 5]
```diff
@@ -270,8 +267,60 @@ message GetAccountAssetTransactions {
 
 Field | Description | Constraint
 -------------- | -------------- | --------------
-Account ID | account id to request transactions from  | `[a-z]{1,9}\@[a-z]{1,9}`
-Asset ID | asset id in order to filter transactions containing this asset | `[a-z]{1,9}\#[a-z]{1,9}`
+Account ID | account id to request transactions from  | username@domain
+Asset ID | asset id in order to filter transactions containing this asset | `[A-Za-z0-9]{1,9}\#[A-Za-z0-9]{1,9}`
```
Comment (l4l) [inline source/includes/_queries.md]: Maybe follow the similar way as for `Account ID`? e.g `asset_name@domain`

**Review by l4l — CHANGES_REQUESTED** [review]

File: source/includes/_queries.md [hunk 6]
```diff
@@ -40,13 +41,22 @@ Each query consists of following things:
     <li> **Query object** might be any of types, described below </li>
```
Comment (nickaleks) [inline source/includes/_queries.md]: "might be of any type, described below" sounds better in my opinion

File: source/includes/_queries.md (hunk 1 above)
Comment (nickaleks) [reply source/includes/_queries.md]: Also, I do not think we use any other signature systems, so it is better to specify that keys are formed using ed25519, and then just say "keys". Moreover, link to ed25519 itself would be good for reference.

File: source/includes/_queries.md (hunk 2 above)
Comment (nickaleks) [reply source/includes/_queries.md]: I would replace "shouldn't" to "should not", as documentation is quite formal, and "from the peer's time" is more grammatically correct.

File: source/includes/_queries.md [hunk 7]
```diff
@@ -40,13 +41,22 @@ Each query consists of following things:
...
 
-**Signatures** contain one or many signatures (ed25519 pubkey + signature):
+**Signatures** contain one or many signatures (ed25519 pubkey + signature).
+
+## Validation
+
+The validation for all commands includes:
+
+ * timestamp — it should be in more than time of the peer minus 24 hours
+ * signature of query creator — is used for checking the identity of query creator
```
Comment (nickaleks) [inline source/includes/_queries.md]: "is" is unnecessary as we already have -

File: source/includes/_queries.md [hunk 8]
```diff
@@ -40,13 +41,22 @@ Each query consists of following things:
...
+ * query counter — is checked to be incremented with every subsequent query from this query creator
```
Comment (nickaleks) [inline source/includes/_queries.md]: "this" is unnecessary

File: source/includes/_queries.md [hunk 9]
```diff
@@ -40,13 +41,22 @@ Each query consists of following things:
...
+ * roles — depending on the role of query creator, it can be applied to the same account, account in the domain, in the whole system or not allowed at all
 
 ## Get account
 
 ### Purpose
 
-Purpose of _get account_ query is to get state of an account.
+Purpose of _get account_ query is to get the state of an account.
 
 Given this Query, in successful case Iroha returns `AccountResponse`, which contains Account object with following fields:
```
Comment (nickaleks) [inline source/includes/_queries.md]: Considering we do not cover successful case, "in successful case" is not needed. Also "Iroha returns" does not sound very good to me. As the query itself is an action, we can say "This Query returns ..." which is shorter.

File: source/includes/_queries.md [hunk 10]
```diff
@@ -173,19 +176,13 @@ Field | Description | Constraint
...
-Stateless validation (check timestamp and signature)
-
 ## Get account transactions
 
 ### Purpose
 
-In case when a list of transactions per user is needed, `GetAccountTransactions` query can be formed.
-
-Given this Query, in successful case Iroha returns `TransactionsResponse`, which is a collection of transactions.
+In a case when a list of transactions per user is needed, `GetAccountTransactions` query can be formed.
```
Comment (nickaleks) [inline source/includes/_queries.md]: "per account" sounds less ambiguous

File: source/includes/_queries.md [hunk 11]
```diff
@@ -236,9 +233,9 @@ Transactions | an array of transactions for given account | Committed transactio
 
 ### Purpose
 
-When an information about transaction per user using certain asset is needed, `GetAccountAssetTransactions` query is sent.
+When an information about transaction per user who is using certain asset is needed, `GetAccountAssetTransactions` query is sent.
```
Comment (nickaleks) [inline source/includes/_queries.md]: As far as I understand, this query returns all transactions for a pair of asset and account. It is very hard to infer that from this sentence. Can we reword that, for example: "`GetAccountAssetTransactions` query returns all transactions associated with given account *and* asset"

File: source/includes/_queries.md [hunk 12]
```diff
@@ -347,21 +396,17 @@ message AccountAssetResponse {
...
-### Validation
-
-Stateless validation (check timestamp and signature)
-
 ## Get asset info
 
 ### Purpose
 
-In order to know precision for given asset, and other related info in future, such as description, etc. user can send `GetAssetInfo` query to Iroha network
+In order to know precision for given asset, and other related info in future, such as description of asset, etc. user can send `GetAssetInfo` query to Iroha network
```
Comment (nickaleks) [inline source/includes/_queries.md]: I think "to Iroha network", is unnecessary, as a reader is unlikely to confuse that with some other blockchain network.

File: source/includes/_queries.md [hunk 13]
```diff
@@ -411,21 +456,17 @@ message Asset {
...
-### Validation
-
-Stateless validation (check timestamp and signature)
-
 ## Get roles
 
 ### Purpose
 
-To get available roles in the system, user can send `GetRoles` query to Iroha network.
+To get available roles in the system, a user can send `GetRoles` query to Iroha network.
```
Comment (nickaleks) [inline source/includes/_queries.md]: "existing" is clearer than "available"

**Review by nickaleks — CHANGES_REQUESTED** [review]
In addition to comments in this pr, after a quick look at the file, I found following issues:

Line 30
Each query consists of following things:
maybe:
Each query consists of the following fields

Line 32
<li> Payload, which contains one created time, id of account creator, query counter and a query of one of possible types </li>
"one" is unnecessary.

File: source/includes/_queries.md (hunk 5 above)
Comment (neewy) [reply source/includes/_queries.md]: maybe I should put description for asset_name and domain separately somewhere then

File: source/includes/_queries.md (hunk 6 above)
Comment (neewy) [reply source/includes/_queries.md]: ok

File: source/includes/_queries.md (hunk 7 above)
Comment (neewy) [reply source/includes/_queries.md]: ok

File: source/includes/_queries.md (hunk 8 above)
Comment (neewy) [reply source/includes/_queries.md]: ok

**Review by nickaleks — APPROVED** [review]

**Review by l4l — APPROVED** [review]
👍

**Pull request merged**